tests/moviePage.spec.js

```diff
@@ -0,0 +1,57 @@
+// Test moviePage function: Only show upcoming screenings
+
+import { jest } from '@jest/globals';
+import moviePage from '../src/moviePage.js';
+
+const cmsAdapter = {
+  async loadScreenings() {
+    // Mock data: Past and future screenings
+    const mockScreenings = [
+      {
+        id: 1,
+        attributes: {
+          start_time: '2024-02-06T12:00:00.000Z', // Past screening
+        },
+      },
+      {
+        id: 2,
+        attributes: {
+          start_time: '2024-02-08T15:00:00.000Z', // Future screening
+        },
+      },
+      {
+        id: 3,
+        attributes: {
+          start_time: '2024-02-10T19:00:00.000Z', // Future screening
+        },
+      },
+    ];
+    return mockScreenings;
+  },
+};
+
+describe('moviePage', () => {
+  test('Test moviePage function: Only show upcoming screenings', async () => {
+    const response = {
+      json: jest.fn(),
+    };
+
+    await moviePage(response, cmsAdapter);
+
+    // Expected result: Only future screenings
+    const expectedResult = {
+      screenings: [
+        {
+          id: 2,
+          time: '2024-02-08 15:00', // Future screening
+        },
+        {
+          id: 3,
+          time: '2024-02-10 19:00', // Future screening
+        },
+      ],
+    };
+
+    expect(response.json).toHaveBeenCalledWith(expectedResult);

```

Comment:
This is a very interesting (but a bit unexpected) approach, where you have designed your `moviePage()` function to handle aspects of output HTTP instead of just returning it's value. This means that you have to verify it's output by "spying" on a mock response object.

The more common approach would to design it in such a way that it returns the result, and to compare the returned result with the expected result.

But it's great to have experimented with `toHaveBeenCalledWith()` and other spying mechanisms.